reference/reference_test.go

```diff
@@ -0,0 +1,129 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package reference_test
+
+import (
+	"bytes"
+	"fmt"
+	"hash"
+	"testing"
+
+	"github.com/ethersphere/bmt/reference"
+	"gitlab.com/nolash/go-mockbytes"
+
+	"golang.org/x/crypto/sha3"
+)
+
+// calculates the hash of the data using hash.Hash
+func doSum(h hash.Hash, b []byte, data ...[]byte) ([]byte, error) {
+	h.Reset()
+	for _, v := range data {
+		var err error
+		_, err = h.Write(v)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return h.Sum(b), nil
+}
+
+// calculates the Keccak256 SHA3 hash of the data
+func sha3hash(t *testing.T, data ...[]byte) []byte {
```

Comment:
It would be good this function as t.Helper.

Reply:
That means just add `t.Helper()` as first line in the function? @janos